spec/tbd_tests_spec.rb

```diff
@@ -1566,3 +1566,414 @@
+                  }
+                },
+                "additionalProperties": false,
+                "required": [
+                  "id",
+                  "count"
+                ]
+              },
+              "uniqueItems": true
+            }
+          },
+          "additionalProperties": false,
+          "minProperties": 2,
+          "required": [
+            "id"
+          ]
+        },
+        "Space": {
+          "description": "Space default PSI-set (optional for OS)",
+          "type": "object",
+          "required": [
+            "id",
+            "psi"
+          ],
+          "properties": {
+            "id": {
+              "title": "e.g. OS space or E+ zone identifier",
+              "type": "string"
+            },
+            "psi": {
+              "title": "PSI-set identifier",
+              "type": "string"
+            }
+          },
+          "additionalProperties": false
+        },
+        "Story": {
+          "title": "Story default PSI-set (optional for OS)",
+          "type": "object",
+          "properties": {
+            "id": {
+              "title": "e.g. OS story identifier",
+              "type": "string"
+            },
+            "psi": {
+              "title": "PSI-set identifier",
+              "type": "string"
+            }
+          },
+          "additionalProperties": false,
+          "required": [
+            "id",
+            "psi"
+          ]
+        },
+        "Unit": {
+          "title": "Building unit default PSI-set (optional for OS)",
+          "type": "object",
+          "properties": {
+            "psi": {
+              "title": "PSI-set identifier",
+              "type": "string"
+            }
+          },
+          "additionalProperties": false,
+          "required": [
+            "psi"
+          ]
+        },
+        "Log": {
+          "title": "TBD log messages",
+          "type": "string"
+        }
+      }
+    }
+
+    tbd_o =
+    {
+      "schema": "https://github.com/rd2/tbd/blob/master/tbd.schema.json",
+      "description": "testing basic JSON validation",
+      "psis": [
+        {
+          "id": "good",
+          "parapet": 0.5
+        },
+        {
+          "id": "compliant",
+          "rimjoist": 0.3,
+          "parapet": 0.325,
+          "fenestration": 0.35,
+          "concave": 0.45,
+          "convex": 0.45,
+          "balcony": 0.5,
+          "party": 0.5,
+          "grade": 0.45
+        }
+      ],
+      "khis": [
+        {
+          "id": "column",
+          "point": 0.5
+        },
+        {
+          "id": "support",
+          "point": 0.5
+        }
+      ],
+      "edges": [
+        {
+          "psi": "compliant",
+          "type": "party",
+          "length": 2.5,
+          "surfaces": [
+            "front wall"
+          ]
+        }
+      ],
+      "surfaces": [
+        {
+          "id": "front wall",
+          "khis": [
+            {
+              "id": "column",
```

Comment:
These khi values just apply to the surface, they don't have to be associated with a physical point?

Reply:
Correct. Their XYZ location along the "front wall" isn't relevant.